src/pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -574,12 +575,12 @@ def __max_ckpt_version_in_folder(dir_path: _PATH, name_key: str = "ckpt_") -> Op
         """
 
         # check directory existence
-        fs = get_filesystem(dir_path)
+        fs, uri = url_to_fs(dir_path)
```

Comment:
@leoleoasd why do we have to `fs.listdir(uri)` but` fs.exists(dir_path)` is ok?
There are many other places in our code base where we use fsspec (modelcheckpiont, saving utilities, tensorboard, profiler, etc.). Does that mean we need to update the usage there too?

Reply:
https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem
According to the document, yes. 
But for mysterious reasons, `exists` works but `listdir` don't work. I'm only changing `listdir` because it blocks my work, and I'm not sure if we have to follow the standard behavior or "if it works, don't touch it". Do we have unit tests for fsspec file systems?

In the document, all functions that aren't bound to a specific FS have parameters named `url` while all methods of `AbstractFileSystem` have parameters named `path`. I think we should update all usages (but not for tensorboard, since tensorboard has its own way of calling fsspec)